File: backend/src/schemas/feed.py
from pydantic import BaseModel

from src.schemas.story import StoryWithCoverage


class PaginationMeta(BaseModel):
    total: int
    limit: int
    offset: int
    has_more: bool


class FeedResponse(BaseModel):
    stories: list[StoryWithCoverage]
    pagination: PaginationMeta

File: mobile/types/feed.ts
import { Story } from "./story";

export interface PaginationMeta {
  total: number;
  limit: number;
  offset: number;
  has_more: boolean;
}

export interface FeedResponse {
  stories: Story[];
  pagination: PaginationMeta;
}

export type FeedFilter = "all" | "blindspot" | "balanced";
